Commit e73326e breaks Gitian build

### Describe the issue
All Gitian builds after https://github.com/zcash/zcash/commit/e73326eed0d987a6a90aac93745059f9be212d77 fail with:

```
In file included from bitcoind.cpp:9:0:
main.h:29:26: fatal error: addressindex.h: No such file or directory
 #include "addressindex.h"
                          ^
compilation terminated.
Makefile:5689: recipe for target 'zcashd-bitcoind.o' failed
make[2]: *** [zcashd-bitcoind.o] Error 1
make[2]: Leaving directory '/home/debian/build/zcash/distsrc-x86_64-unknown-linux-gnu/src'
Makefile:5737: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/home/debian/build/zcash/distsrc-x86_64-unknown-linux-gnu/src'
Makefile:615: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
```

### Can you reliably reproduce the issue?
#### If so, please list the steps to reproduce below:
1. Destroy any existing zcash-builder VMs
2. Set `ZCASH_VERSION` to e73326eed0d987a6a90aac93745059f9be212d77 or a later commit
3. Run `vagrant up zcash-build`
4. Once built `vagrant ssh zcash-build`
5. Run `./gitian-build.sh`
6. `gitian-builder/var/build.log` contains the following error and build does not complete

### Expected behaviour
It should build as the previous commit does https://github.com/zcash/zcash/commit/926fd0ce2836fd665a956f52566824ecbe143c47

### The version of Zcash you were using:
Vagrant 2.2.4
debian/stretch64 (virtualbox, 9.8.0)

Discussion from here: https://github.com/zcash/zcash/pull/3951

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -114,6 +114,7 @@ LIBZCASH_H = \
 .PHONY: FORCE collate-libsnark check-symbols check-security
 # bitcoin core #
 BITCOIN_CORE_H = \
+  addressindex.h \
   addrman.h \
   alert.h \
   amount.h \

